dev/fattest.simplicity/src/componenttest/rules/repeater/FeatureReplacementAction.java

```diff
@@ -495,9 +497,70 @@ public FeatureReplacementAction withBeta() {
         return this;
     }
 
+    @Override
+    public RepeatTestAction checkIfFeaturesPresent(boolean checkIfFeaturesPresent) {
+        this.checkFeaturesPresent = checkIfFeaturesPresent;
+        return this;
+    }
+
     @Override
     public boolean isEnabled() {
-        return checkEnabled();
+        boolean enabled = checkEnabled();
+        if (enabled && checkFeaturesPresent) {
+            enabled = checkAllFeatureFilesPresent();
+        }
+        return enabled;
+    }
+
+    public boolean checkAllFeatureFilesPresent() {
+        Bootstrap b;
+        try {
+            b = Bootstrap.getInstance();
+        } catch (Exception e) {
+            Log.info(c, "checkAllFeatureFilesPresent", "Skipping action '" + toString() + "' because Bootstrap.getInstance() failed.");
+            return false;
+        }
+        String installRoot = b.getValue("libertyInstallPath");
+
+        Set<String> featuresToCheck = new HashSet<String>();
+        featuresToCheck.addAll(addFeatures);
+        featuresToCheck.addAll(alwaysAddFeatures);
+
+        final String path = installRoot + "/lib/features/";
+        final String oldPrefix = "com.ibm.websphere.appserver.";
+        final String newPrefix = "io.openliberty.";
+        final String newPrefixWithJakarta = "io.openliberty.jakarta.";
+        final String suffix = ".mf";
+
+        //We occasionally create features that only exist for fat tests and will not always be present in the liberty image.
+        // To account for this we'll tolerate a 20% failure ratio.
+        int failures = 0;
+        String skipMessage = "Skipping action '" + toString() + "' because ";
+
+        for (String feature : featuresToCheck) {
+
+            File oldFileLocation = new File(path + oldPrefix + feature + suffix);
+            File newFileLocation = new File(path + newPrefix + feature + suffix);
+            File newFileJakartaLocation = new File(path + newPrefixWithJakarta + feature + suffix);
+            File noPrefixFileLocation = new File(path + feature + suffix);
+
+            if (!oldFileLocation.exists() && !newFileLocation.exists() && !noPrefixFileLocation.exists()) {
+                skipMessage = skipMessage + System.lineSeparator() + feature + " was not found in any of: "
+                              + System.lineSeparator() + oldFileLocation.getAbsolutePath()
+                              + System.lineSeparator() + newFileLocation.getAbsolutePath()
+                              + System.lineSeparator() + newFileJakartaLocation.getAbsolutePath()
+                              + System.lineSeparator() + noPrefixFileLocation.getAbsolutePath();
+                failures++;
+            }
+        }
```

Comment:
Unsure whether checking like this is a good idea.

I think the name of the manifest file is just a convention, to really know what feature it is we'd have to look at the `Subsystem-SymbolicName` header, though since `FeatureReplacementAction` works with short names, checking `IBM-ShortName` may also make sense.

Reply:
We could build up a giant set of all IBM short names by actually reading the mf files